src/jaxsim/api/model.py

```diff
@@ -1931,11 +1932,17 @@ def step(
         ),
     )
 
+    exceptions.raise_if(
+        condition=t0_ns + jnp.array(dt * 1e9).astype(t0_ns.dtype) < t0_ns,
```

Comment:
```suggestion
        condition=t0_ns + jnp.array(dt * 1e9, dtype=t0_ns.dtype) < t0_ns,
```

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/225/commits/051704a3690fb977a1f6e666b7846da90be5fb02